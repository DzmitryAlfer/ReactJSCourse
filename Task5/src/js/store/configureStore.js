import { createStore, applyMiddleware, compose } from 'redux'
import { createLogger } from 'redux-logger'
import createSagaMiddleware, { END } from 'redux-saga'
import rootReducer from '../reducers'
import { persistStore, persistReducer } from 'redux-persist'
import storage from 'redux-persist/lib/storage'

const persistConfig = {
    key: 'root',
    storage,
};

const persistedReducer = persistReducer(persistConfig, rootReducer)

export default function configureStore(initialState) {
    const sagaMiddleware = createSagaMiddleware()

    const store = createStore(
        persistedReducer,
        initialState,
        compose(
            applyMiddleware(
                sagaMiddleware,
                createLogger()
            )
        )
    );
    const persistor = persistStore(store);

    /*if (module.hot) {
        // Enable Webpack hot module replacement for reducers
        module.hot.accept('../reducers', () => {
            const nextRootReducer = require('../reducers').default;
            store.replaceReducer(nextRootReducer);
        });
    }*/

    store.runSaga = sagaMiddleware.run;
    store.close = () => store.dispatch(END);
    return {store, persistor};
}