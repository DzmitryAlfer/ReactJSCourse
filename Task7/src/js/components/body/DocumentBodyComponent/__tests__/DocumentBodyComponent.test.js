import React from 'react';
import Renderer from 'react-test-renderer';
import { DocumentBodyComponent } from '../';
import { Provider } from 'react-redux';
import configureStore from '../../../../store/configureStore';

const { store } = configureStore();

jest.mock('../../search/SearchResultsContainer/index');

describe('DocumentBodyComponent', () => {
    test('Component render', () => {
        const tree = Renderer.create(<Provider store={store}>
            <DocumentBodyComponent />
        </Provider>).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
