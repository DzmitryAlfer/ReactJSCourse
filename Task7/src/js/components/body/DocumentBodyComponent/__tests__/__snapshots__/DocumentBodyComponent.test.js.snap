// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentBodyComponent Component render 1`] = `
<main
  className="main-body"
/>
`;
